apps/dashboard/src/components/engine/configuration/kms-gcp-config.tsx

```diff
@@ -1,165 +1,220 @@
+              form.getFieldState(
+                "gcpApplicationCredentialEmail",
+                form.formState,
+              ).error?.message
+            }
+            htmlFor="gcp-credential-email"
+            isRequired={false}
+            tooltip={null}
+          >
             <Input
-              placeholder="Your Google App Credential Email"
-              autoComplete="off"
+              id="gcp-credential-email"
               type="text"
-              {...form.register("gcpApplicationCredentialEmail", {
-                required: true,
-              })}
+              placeholder="service-account@my-gcp-project.iam.gserviceaccount.com"
+              autoComplete="off"
+              {...form.register("gcpApplicationCredentialEmail")}
             />
-          </FormControl>
-        </Flex>
-        <FormControl isRequired>
-          <FormLabel>Private Key</FormLabel>
-          <Input
-            w={{ base: "full", md: "49%" }}
-            placeholder="Your Google App Private Key"
+          </FormFieldSetup>
+        </div>
+
+        <FormFieldSetup
+          label="Private Key"
+          errorMessage={
+            form.getFieldState(
+              "gcpApplicationCredentialPrivateKey",
+              form.formState,
+            ).error?.message
+          }
+          htmlFor="gcp-private-key"
+          isRequired={false}
+          tooltip={null}
+        >
+          <Textarea
+            id="gcp-private-key"
+            placeholder={
+              "-----BEGIN PRIVATE KEY-----\n...\n-----END PRIVATE KEY-----"
+            }
             autoComplete="off"
-            type="text"
-            {...form.register("gcpApplicationCredentialPrivateKey", {
-              required: true,
-            })}
+            {...form.register("gcpApplicationCredentialPrivateKey")}
           />
-        </FormControl>
-      </Card>
-      <Flex justifyContent="end" gap={4} alignItems="center">
-        {form.formState.isDirty && (
-          <Text color="red.500">
-            This will reset your other backend wallet configurations
-          </Text>
-        )}
-        <Button
-          isDisabled={!form.formState.isDirty}
-          w={{ base: "full", md: "inherit" }}
-          colorScheme="primary"
-          px={12}
-          type="submit"
-        >
-          {form.formState.isSubmitting ? "Saving..." : "Save"}
-        </Button>
-      </Flex>
-    </Flex>
+          <FormDescription className="pt-2">
+            This will not be shown again.
+          </FormDescription>
+        </FormFieldSetup>
+
+        <div className="flex items-center justify-end gap-4">
+          {!supportsMultipleWalletTypes && (
+            <p className="text-destructive-text text-sm">
+              This will clear other credentials.
+            </p>
+          )}
+          <Button disabled={!form.formState.isDirty} type="submit">
+            {form.formState.isSubmitting ? "Saving..." : "Save"}
```

Comment:
same here - setGcpKmsConfig is not awaited in onSubmit - you won't get correct `isSubmitting` state - prefer using `mutation.isPending` directly or add await